### PR TITLE
Support filter image by reference to the image name

### LINKED ISSUE #2266: Allow filtering images by reference
/kind feature

**Description**

Docker CLI allows to filter by reference and automatically filters by reference if given an additional parameter:

```
$ sudo docker images --filter reference='docker.io/*'
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
docker.io/centos    7.4.1708            295a0b2bd8ea        3 months ago        197 MB
$ sudo docker images 'docker.io/*'
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
docker.io/centos    7.4.1708            295a0b2bd8ea        3 months ago        197 MB
```

Podman currently does not implement [this](https://github.com/containers/libpod/blob/b1d4dbd335f4801487ef86964b6833c3604ceef5/cmd/podman/images.go#L375)

This can surface if users are using the Docker CLI compatibility shim with scripts (as I was doing).

### PR DESCRIPTION
**Pull request by QiWang19**
fix #2266 
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
needs an integration test or two/ else lgtm

**Comment by TomSweeneyRedHat** [comment]
ditto @baude's comment.  LGTM once new tests are provided.

**Comment by rhatdan** [comment]
@QiWang19 and I just did a lot of rework on this one.  She will be pushing a new PR along with tests.

File: test/e2e/images_test.go [hunk 1]
```diff
@@ -105,6 +105,16 @@ var _ = Describe("Podman images", func() {
 		Expect(len(session.OutputToStringArray())).To(Equal(1))
 	})
 
+	It("podman images filter reference", func() {
+		if podmanTest.RemoteTest {
+			Skip("Does not work on remote client")
+		}
+		result := podmanTest.Podman([]string{"images", "-q", "-f", "reference=docker.io*"})
```
Comment (rhatdan) [inline test/e2e/images_test.go]: Lets add a couple of others.

Look for only alpine, make sure you get one
Look for a*pine a make sure you get one.
and look for bogus.  and make sure you get none.

**Comment by mheon** [comment]
Needs to update RELEASE_NOTES.md with changes

**Comment by mheon** [comment]
Nevermind the noise, no need to hit the release notes doc

File: test/e2e/images_test.go (hunk 1 above)
Comment (QiWang19) [reply test/e2e/images_test.go]: fixed :raising_hand_woman:

**Comment by rhatdan** [comment]
@QiWang19 Did you add tests?

**Comment by QiWang19** [comment]
@rhatdan added tests. Ready to be merged.

**Pull request merged**